pkg/reconciler/taskrun/taskrun.go

```diff
@@ -203,28 +204,40 @@ func (c *Reconciler) updateStatusLabelsAndAnnotations(tr, original *v1alpha1.Tas
 	return nil
 }
 
-func (c *Reconciler) getTaskFunc(tr *v1alpha1.TaskRun) (resources.GetTask, v1alpha1.TaskKind) {
-	var gtFunc resources.GetTask
-	kind := v1alpha1.NamespacedTaskKind
+func (c *Reconciler) getTaskFunc(tr *v1alpha1.TaskRun) (resources.GetTask, v1alpha1.TaskKind, error) {
```

Comment:
this is scope creep from me but something that could make this function way better would be to move it into another package and not having it use the reconciler. this would mean passing c.kubeclientset in directly, but it would also mean we could have unit tests for it directly which would be great! `GetResourceFromBinding` is a very (very) similar example:

https://github.com/tektoncd/pipeline/blob/2e7e68f2fba58f90955c369123b228ab5654190e/pkg/reconciler/taskrun/resources/taskresourceresolution.go#L78-L80